spec/models/manageiq/providers/base_manager/metrics_capture_spec.rb

```diff
@@ -20,41 +20,39 @@
     end
   end
 
-  describe ".perf_capture_now" do
-    context "with enabled and disabled targets" do
-      before do
-        MiqRegion.seed
-        storages = FactoryBot.create_list(:storage_target_vmware, 2)
-        clusters = FactoryBot.create_list(:cluster_target, 2)
-        ems.ems_clusters = clusters
-
-        6.times do |n|
-          host = FactoryBot.create(:host_target_vmware, :ext_management_system => ems)
-          ems.hosts << host
-
-          clusters[n / 2].hosts << host if n < 4
-          host.storages << storages[n / 3]
-        end
-
-        MiqQueue.delete_all
-      end
-
-      context "executing perf_capture_gap" do
-        before do
-          t = Time.now.utc
-          Metric::Capture.perf_capture_gap(t - 7.days, t - 5.days, nil, ems.id)
-        end
-
-        it "should queue up enabled targets for historical" do
-          expect(MiqQueue.count).to eq(10)
-
-          expected_targets = Metric::Targets.capture_ems_targets(ems.reload, :exclude_storages => true)
-          expected = expected_targets.flat_map { |t| [[t, "historical"]] * 2 } # Vm, Host, Host, Vm, Host
+  describe ".perf_capture_gap" do
+    let(:host) { FactoryBot.create(:host_vmware, :ext_management_system => ems, :perf_capture_enabled => true) }
+    let(:vm)   { FactoryBot.create(:vm_vmware, :ext_management_system => ems, :host => host) }
+    let(:host2) do
+      FactoryBot.create(
+        :host_vmware,
+        :ext_management_system => ems,
+        :perf_capture_enabled  => true,
+        :storages              => [storage],
+        :ems_cluster           => FactoryBot.create(:ems_cluster, :perf_capture_enabled => true, :ext_management_system => ems)
+      )
+    end
+    let(:vm2)     { FactoryBot.create(:vm_vmware, :ext_management_system => ems, :host => host2) }
+    let(:host3)   { FactoryBot.create(:host_vmware, :ext_management_system => ems, :perf_capture_enabled => true) }
+    let(:storage) { FactoryBot.create(:storage_vmware, :perf_capture_enabled => true) }
 
-          selected = queue_intervals(MiqQueue.all)
+    it "should queue up targets for historical" do
+      MiqRegion.seed
+      vm
```

Comment:
If this is just to instantiate the `let` can you use `let!` instead? (same for vm2, host3)